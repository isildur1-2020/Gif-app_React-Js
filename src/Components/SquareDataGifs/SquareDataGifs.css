.SquareDataGrid {
  display: flex;
  flex-wrap: wrap;
}
