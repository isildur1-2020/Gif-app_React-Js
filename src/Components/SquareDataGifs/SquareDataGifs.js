import React from "react";
import PropTypes from "prop-types";
import SquareItem from "../SquareItem/SquareItem";
import { useDataGifs } from "../../hooks/useDataGifs";
import "./SquareDataGifs.css";

const SquareData = ({ word }) => {
  const { error, loading, data } = useDataGifs(word);

  return (
    <>
      {loading ? (
        <h2>Cargando...</h2>
      ) : !error ? (
        <div className="SquareDataGrid">
          {data &&
            data.map(({ id, title, url }) => (
              <SquareItem key={id} title={title} url={url} />
            ))}
        </div>
      ) : (
        <h1>No tenemos resultados...</h1>
      )}
    </>
  );
};

SquareData.propTypes = {
  word: PropTypes.string.isRequired,
};

export default SquareData;
