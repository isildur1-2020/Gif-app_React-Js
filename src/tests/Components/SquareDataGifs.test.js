import React from "react";
import { shallow } from "enzyme";
import SquareDataGifs from "../../Components/SquareDataGifs/SquareDataGifs";

describe("Pruebas en <SquareDataGifs />", () => {
  test("Debe mostrar el componente correctamente", () => {
    // ARRANGE
    const wrapper = shallow(<SquareDataGifs word="name" />);
    // ACT
    // ASSERT
    expect(wrapper).toMatchSnapshot();
  });
});
