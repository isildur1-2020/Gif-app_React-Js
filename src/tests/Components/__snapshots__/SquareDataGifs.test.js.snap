// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SquareDataGifs /> Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h2>
    Cargando...
  </h2>
</Fragment>
`;
